# Downloading NGS data

In order to download NGS data from the Sequence Read Archive (SRA) you will need to use the [***SRA Tools***](https://github.com/ncbi/sra-tools) package .

```
conda deactivate  # If you are not at (base) environment
conda create --name sratools
conda activate sratools
conda install bioconda::sra-tools
```

Next you will need to set up the the package to be able to access public and controlled-access data in the cloud, running:

```
vdb-config -i
```

Then Enable Remote Access (e) in the Main (m) tab and navigante to Cache (c) to enable local file-sharing (i), exit (x), saving the changes (y) and confirm (o). [Other settings](https://github.com/ncbi/sra-tools/wiki/03.-Quick-Toolkit-Configuration) can be changed later by running the same command above.


Make a folder to hold your data for the MARRIO Genomics Course, let's say:

```
mkdir /mnt/c/MARRIO_genomics
```

In this tutorial we will focus on retrieving the WGS data from *Hymedesmia metanophila* ([SRR7464845](https://www.ncbi.nlm.nih.gov/sra/?term=SRR7464845)) generated by [Rubin-Blum et al. (2019)](https://doi.org/10.1038/s41396-019-0346-7) and used by [Leite et al. (2024)](https://doi.org/10.1007/s12041-024-01485-7) to recover its mtDNA and rDNA.

To download this sequecing experiment from NCBI servers, viz., its FASTQ files with raw PE reads, you can run the command:

```
fasterq-dump SRR7464845 -O /mnt/c/MARRIO_genomics/SRR7464845 --split-files --threads 8 --progress
```

After some time, you will get the output files `SRR7464845_1.fastq` and `SRR7464845_2.fastq` files, each with 7.63 Gb.

Now that you have a NGS data, you can apply [filtering and quality control](https://github.com/depaulats/MARRIO_genomics/blob/main/clean.md) analyses.

Or you can go back to the [tutorials](https://github.com/depaulats/MARRIO_genomics/blob/main/tutorials.md). 
