# Mapping reads against a reference

In order to map your filtered reads against a reference you will need to use the [BBMap](https://sourceforge.net/projects/bbmap/) package.

You do not need an environment to run BBMap since it "is written in pure Java, can run on any platform, and has no dependencies other than Java being installed".

Thus, just download the package and extract it to `/mnt/c/MARRIO_genomics/`:

```
wget https://sourceforge.net/projects/bbmap/files/BBMap_39.33.tar.gz -P /mnt/c/MARRIO_genomics/
tar -xvzf /mnt/c/MARRIO_genomics/BBMap_39.33.tar.gz -C /mnt/c/MARRIO_genomics/
rm /mnt/c/MARRIO_genomics/BBMap_39.33.tar.gz
```

Before you can proceed, you will need to dowanload the FASTA files from both mtDNA () and rDNA () to use as references. These sequences correspond to the mitochondrial genome and nuclear 45S rDNA cluster from *Hymedesmia methanophila* generated by [Leite et al. (2024)](https://doi.org/10.1007/s12041-024-01485-7). If you want to conduct your analysis with the original references used in the study, check their accessions numbers in the paper and download them from GenBank as an exercise.

To download the mtDNA (`BK068137.fa`) and the rDNA (`BK068393.fa`) references, run:

```
wget https://raw.githubusercontent.com/depaulats/MARRIO_genomics/main/reference/BK068137.fa  -P /mnt/c/MARRIO_genomics/
wget https://raw.githubusercontent.com/depaulats/MARRIO_genomics/main/reference/BK068393.fa  -P /mnt/c/MARRIO_genomics/
```

Now, to map the reads against the mtDNA reference, run the comand:

```
/mnt/c/MARRIO_genomics/bbmap/bbmap.sh \
  ref=/mnt/c/MARRIO_genomics/BK068137.fa nodisk \
  in1=/mnt/c/MARRIO_genomics/SRR7464845/SRR7464845_1_paired.fastq \
  in2=/mnt/c/MARRIO_genomics/SRR7464845/SRR7464845_1_paired.fastq \
  mappedonly=t \
  out=/mnt/c/MARRIO_genomics/SRR7464845/SRR7464845_mtDNA_mapped.fastq
```

Now, to map the reads against the rDNA reference, you will have to edit the command above. I'm sure you can do it!
